shell/platform/embedder/tests/embedder_unittests.cc

```diff
@@ -4207,5 +4207,69 @@ TEST_F(EmbedderTest, CompositorRenderTargetsAreInStableOrder) {
   latch.Wait();
 }
 
+TEST_F(EmbedderTest, InvalidAOTDataSourcesMustReturnError) {
+  FlutterEngineAOTDataSource data_in;
+  FlutterEngineAOTData data_out;
+
+  // Invalid FlutterEngineAOTDataSourceType type specified.
+  ASSERT_EQ(FlutterEngineCreateAOTData(&data_in, &data_out), kInvalidArguments);
+
+  data_in.type = kFlutterEngineAOTDataSourceTypeElfPath;
+  data_in.elf_path = nullptr;
+
+  // Invalid ELF path specified.
+  ASSERT_EQ(FlutterEngineCreateAOTData(&data_in, &data_out), kInvalidArguments);
+
+  data_in.elf_path = "";
+
+  // Invalid ELF path specified.
+  ASSERT_EQ(FlutterEngineCreateAOTData(&data_in, &data_out), kInvalidArguments);
+
+  data_in.elf_path = "/dev/null";
+
+  // Could not read ELF file.
+  ASSERT_EQ(FlutterEngineCreateAOTData(&data_in, &data_out), kInvalidArguments);
+}
```

Comment:
There is no test that explicitly checks the collect call (it is only being exercised in the fixture). I would add one case here that successfully loads and then unloads an ELF data source.